atlp-rwanda/e-commerce-serge-30-bn
# feat:product search functionality[#187364878]

- Enable users to search for products by name, price range, category, or a combination of these criteria.
- Implement search functionality with AND operation to allow users to refine their search results effectively.

[Delivers #187364878]

## ℹ️ Description

This pull request enables users to search for products using various criteria such as name, price range, category, or a combination thereof. Additionally, it implements search functionality with an AND operation, allowing users to refine their search results effectively.

Fixes #(issue)

## 🧪 How can this be tested?

- Search by Name: Manually enter a product name in the search field and verify that products matching the name are displayed in the search results.
- Search by Price Range: Enter a price range in the search field and verify that products within that price range are displayed in the search results.
- Search by Category: Select a category  enter it in the search field and verify that products belonging to that category are displayed in the search results.
- Combination Search: Enter multiple criteria (e.g., name, price range, and category) in the search field and verify that products meeting all criteria are displayed in the search results.
- Create Jest test cases to cover different aspects of the search functionality. This may include testing:
  * Searching by name, price range, category, and combinations thereof.
  * Ensuring that the search results are accurate and relevant.
  * Testing edge cases, such as empty searches or searches with invalid input.

## 📍 Type of change

Please check the options that are relevant.


## ✅ Checklist:
